components/expression/expression-renderer.js

```diff
@@ -172,21 +211,66 @@ genotet.ExpressionRenderer.prototype.drawExpressionMatrix_ = function() {
  * @private
  */
 genotet.ExpressionRenderer.prototype.drawMatrixCells_ = function() {
-  // TODO(liana): Implement this...
-  var cells = this.svgHeatmapContent_.selectAll('rect.cell').data([{
-    width: this.heatmapWidth_,
-    height: this.heatmapHeight_
-  }]);
-  cells.enter().append('rect')
-    .classed('cell', true);
+  var heatmapData = this.data.matrix;
+  var cellWidth = this.heatmapWidth_ / heatmapData.conditionNames.length;
+  var cellHeight = this.heatmapHeight_ / heatmapData.geneNames.length;
+  //console.log(heatmapData);
+  var colorScale = d3.scale.linear()
+    .domain([heatmapData.valueMin, heatmapData.valueMax])
+    .range(genotet.data.redYellowScale);
+
+  this.svgHeatmapContent_.selectAll('g').remove();
+  var cells = this.svgHeatmapContent_.selectAll('rect.cell').data(heatmapData.values);
+  var heatmapRows = cells.enter().append('g')
+    .attr("transform", genotet.utils.getTransform([
+      0,
+      this.conditionLabelHeight_ + this.profileHeight_
+    ]));
   cells.exit().remove();
-  cells
-    .attr('width', function(cell) {
-      return cell.width;
+  var heatmapRects = heatmapRows.selectAll('.rect')
+    .data(_.identity)
+    .enter().append('rect')
+    .attr('width', cellWidth)
+    .attr('height', cellHeight)
+    .attr('x', function(d, i, j) {
+      return i * cellWidth;
+    })
+    .attr('y', function(d, i, j) {
+      return j * cellHeight;
+    })
+    .style('stroke', function(d) {
+      return colorScale(d);
+    })
+    .style('fill', function(d) {
+      return colorScale(d);
     })
-    .attr('height', function(cell) {
-      return cell.height;
-    });
+    .on('mouseover', function(d, i, j) {
+      var hoverCell = d3.event.target;
+      this.signal(
+        'cellHover',
+        [
+          hoverCell,
```

Comment:
Create an object for this and name the attributes. Otherwise you will have to do data[0], data[1] in signal handler, which is confusing.